File: http/src/main/java/webconnect/com/webconnect/WebParam.java
package webconnect.com.webconnect;

import android.app.Dialog;
import android.content.Context;

import java.io.File;
import java.io.Serializable;
import java.util.LinkedHashMap;
import java.util.Map;

import webconnect.com.webconnect.listener.AnalyticsListener;
import webconnect.com.webconnect.listener.OnWebCallback;
import webconnect.com.webconnect.listener.ProgressListener;


/**
 * The type Web param.
 */
public class WebParam implements Serializable {
    Context context;
    String url, baseUrl;
    HttpType httpType = HttpType.GET;
    Map<String, ?> requestParam = new LinkedHashMap<>();
    Map<String, String> queryParam = new LinkedHashMap<>();
    Map<String, String> multipartParam = new LinkedHashMap<>();
    Map<String, File> multipartParamFile = new LinkedHashMap<>();
    Map<String, String> headerParam = new LinkedHashMap<>();
    OnWebCallback callback;
    ProgressListener progressListener;
    AnalyticsListener analyticsListener;
    Dialog dialog;
    Class<?> model = Object.class;
    Class<?> error = Object.class;
    int taskId;
    long connectTimeOut = 0, readTimeOut = 0;
    boolean isCacheEnabled = false, debug = false;
    File file;
    long requestBodyContentlength = -1;

    public enum HttpType {
        GET,
        POST,
        PUT,
        PATCH,
        DELETE,
        HEAD,
        OPTIONS,
        MULTIPART,
        DOWNLOAD
    }
}

File: http/src/main/java/webconnect/com/webconnect/WebParam.kt
package webconnect.com.webconnect

import android.app.Dialog
import android.content.Context
import webconnect.com.webconnect.listener.*
import webconnect.com.webconnect.model.ErrorModel
import webconnect.com.webconnect.model.SuccessModel
import java.io.File
import java.io.Serializable
import java.util.*


/**
 * The type Web param.
 */
class WebParam : Serializable {
    internal var context: Context? = null
    internal var url: String = ""
    internal var baseUrl: String = ""
    internal var httpType = HttpType.GET
    internal var requestParam: Map<String, Any> = LinkedHashMap()
    internal var queryParam: QueryMap<String, String> = QueryMap()
    internal var query: Map<String, String> = LinkedHashMap()
    internal var multipartParam: Map<String, String> = LinkedHashMap()
    internal var multipartParamFile: Map<String, File> = LinkedHashMap()
    internal var multipartParamListFile: Map<String, List<File>> = LinkedHashMap()
    internal var headerParam: Map<String, String> = LinkedHashMap()
    internal var success: OnSuccessListener<Any>? = null
    internal var err: OnErrorListener<Any>? = null
    internal var failure: OnFailureListener? = null
    internal var responseListener: ResponseListener? = null
    internal var progressListener: ProgressListener? = null
    internal var analyticsListener: AnalyticsListener? = null
    internal var loaderListener: LoaderListener? = null
    internal var dialog: Dialog? = null
    internal var model: Class<*> = Any::class.java
    internal var error: Class<*> = Any::class.java
    internal var taskId: Int = 0
    internal var connectTimeOut: Long = 0L
    internal var readTimeOut: Long = 0L
    internal var isCacheEnabled = false
    internal var debug = false
    internal var isJson = false
    internal var file: File? = null
    internal var requestBodyContentlength: Long = -1L

    internal enum class HttpType {
        GET,
        POST,
        PUT,
        PATCH,
        DELETE,
        HEAD,
        OPTIONS,
        MULTIPART,
        DOWNLOAD
    }
}
